### PR TITLE
Feature/split program program package query

### PR DESCRIPTION
**Pull request by Ponchimeow**

### CODE REVIEW

**Comment by Ponchimeow** [comment]
[(2)【課程組合、課程】load不出來 [0]](https://trello.com/c/sfuHVWd9/3010-2%E3%80%90%E8%AA%B2%E7%A8%8B%E7%B5%84%E5%90%88%E3%80%81%E8%AA%B2%E7%A8%8B%E3%80%91load%E4%B8%8D%E5%87%BA%E4%BE%86-0)

File: src/hooks/programPackage.ts [hunk 1]
```diff
@@ -265,48 +265,33 @@ export const useDeliverProgramCollection = () => {
...
-export const useGetProgramPackageCollection = (appId: string) => {
-  const { loading, error, data, refetch } = useQuery<hasura.GET_PROGRAM_PACKAGES, hasura.GET_PROGRAM_PACKAGESVariables>(
+export const useGetProgramPackageCollection = () => {
+  const { loading, error, data, refetch } = useQuery<hasura.GET_PROGRAM_PACKAGES>(
     gql`
-      query GET_PROGRAM_PACKAGES($appId: String!) {
-        program_package(where: { app: { id: { _eq: $appId } } }) {
+      query GET_PROGRAM_PACKAGES {
+        program_package {
           id
```
Comment (emily40830) [inline src/hooks/programPackage.ts:273]: `useGetProgramPackageCollection` is not a suitable naming for data hook.
You can deprecate it.
is it possible to combine `useProgramPackageCollection`data hook ?

**Review by emily40830 — APPROVED** [review]

File: src/hooks/programPackage.ts [hunk 2]
```diff
@@ -12,29 +12,64 @@ export const useProgramPackageCollection = () => {
...
         }))
 
+  if (enrollmentData) {
+    enrollmentData.program_package.forEach(v => {
+      programPackages.forEach(programPackage => {
+        if (programPackage.id === v.id) {
+          programPackage.programPackageEnrollment = sum(
+            v.program_package_plans.map(w => w.program_package_plan_enrollments_aggregate.aggregate?.count || 0),
+          )
+        }
```
Comment (emily40830) [inline src/hooks/programPackage.ts:63]: NITS

```
programPackages.forEach((programPackage)=>{
  const targetEnrollments = enrollmentData.program_package.filter(v=>v.id === programPackage.id)
  // do something for updating each programPackage enrollment from targetEnrollments
  // ....
})

```

File: src/hooks/programPackage.ts [hunk 3]
```diff
@@ -265,58 +300,6 @@ export const useDeliverProgramCollection = () => {
...
-    loading,
-    error,
-    programPackages,
-    refetch,
-  }
-}
-
 export const useGetProgramPackage = (id: string) => {
   const { loading, error, data, refetch } = useQuery<hasura.GET_PROGRAM_PACKAGE, hasura.GET_PROGRAM_PACKAGEVariables>(
     gql`
```
Comment (emily40830) [inline src/hooks/programPackage.ts:305]: NOTE: same issue
useGetProgramPackage => useProgramPackage

File: src/pages/ProgramCollectionAdminPage.tsx [hunk 4]
```diff
@@ -343,18 +343,45 @@ const useProgramPreviewCollection = (
...
+  )
+
+  if (enrollmentData) {
+    enrollmentData.program.forEach(v => {
+      programPreviews.forEach(programPreview => {
+        if (programPreview.id === v.id) {
+          programPreview.enrollment = sum(
+            v.program_plans.map(w => w.program_plan_enrollments_aggregate.aggregate?.count || 0),
+          )
+        }
```
Comment (emily40830) [inline src/pages/ProgramCollectionAdminPage.tsx:380]: same issue

**Review by emily40830 — CHANGES_REQUESTED** [review]

**Review by emily40830 — APPROVED** [review]

**Pull request merged**